File: Source/AlphaTab.Test/Importer/MusicXmlImporterSamplesTest.cs
using System.IO;
using System.Linq;
using System.Runtime.CompilerServices;
using AlphaTab.Model;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace AlphaTab.Test.Importer
{
    [TestClass]
    public class MusicXmlImporterSamplesTests : MusicXmlImporterTestBase
    {
        [TestMethod, AsyncTestMethod]
        public void Test_BeetAnGeSample()
        {
            TestReferenceFile("TestFiles/MusicXmlSamples/BeetAnGeSample.xml");
        }

        [TestMethod, AsyncTestMethod]
        public void Test_Binchois()
        {
            TestReferenceFile("TestFiles/MusicXmlSamples/Binchois.xml");
        }

        [TestMethod, AsyncTestMethod]
        public void Test_BrahWiMeSample()
        {
            TestReferenceFile("TestFiles/MusicXmlSamples/BrahWiMeSample.xml");
        }

        [TestMethod, AsyncTestMethod]
        public void Test_BrookeWestSample()
        {
            TestReferenceFile("TestFiles/MusicXmlSamples/BrookeWestSample.xml");
        }

        [TestMethod, AsyncTestMethod]
        public void Test_Chant()
        {
            TestReferenceFile("TestFiles/MusicXmlSamples/Chant.xml");
        }

        [TestMethod, AsyncTestMethod]
        public void Test_DebuMandSample()
        {
            TestReferenceFile("TestFiles/MusicXmlSamples/DebuMandSample.xml");
        }

        [TestMethod, AsyncTestMethod]
        public void Test_Dichterliebe01()
        {
            TestReferenceFile("TestFiles/MusicXmlSamples/Dichterliebe01.xml");
        }

        [TestMethod, AsyncTestMethod]
        public void Test_Echigo()
        {
            TestReferenceFile("TestFiles/MusicXmlSamples/Echigo.xml");
        }

        [TestMethod, AsyncTestMethod]
        public void Test_FaurReveSample()
        {
            TestReferenceFile("TestFiles/MusicXmlSamples/FaurReveSample.xml");
        }

        [TestMethod, AsyncTestMethod]
        public void Test_MahlFaGe4Sample()
        {
            TestReferenceFile("TestFiles/MusicXmlSamples/MahlFaGe4Sample.xml");
        }

        [TestMethod, AsyncTestMethod]
        public void Test_MozaChloSample()
        {
            TestReferenceFile("TestFiles/MusicXmlSamples/MozaChloSample.xml");
        }

        [TestMethod, AsyncTestMethod]
        public void Test_MozartPianoSonata()
        {
            TestReferenceFile("TestFiles/MusicXmlSamples/MozartPianoSonata.xml");
        }

        [TestMethod, AsyncTestMethod]
        public void Test_MozartTrio()
        {
            TestReferenceFile("TestFiles/MusicXmlSamples/MozartTrio.xml");
        }

        [TestMethod, AsyncTestMethod]
        public void Test_MozaVeilSample()
        {
            TestReferenceFile("TestFiles/MusicXmlSamples/MozaVeilSample.xml");
        }

        [TestMethod, AsyncTestMethod]
        public void Test_Saltarello()
        {
            TestReferenceFile("TestFiles/MusicXmlSamples/Saltarello.xml");
        }

        [TestMethod, AsyncTestMethod]
        public void Test_SchbAvMaSample()
        {
            TestReferenceFile("TestFiles/MusicXmlSamples/SchbAvMaSample.xml");
        }

        [TestMethod, AsyncTestMethod]
        public void Test_Telemann()
        {
            TestReferenceFile("TestFiles/MusicXmlSamples/Telemann.xml");
        }
    }
}

File: test/importer/MusicXmlImporterSamples.test.ts
import { MusicXmlImporterTestHelper } from '@test/importer/MusicXmlImporterTestHelper';

describe('MusicXmlImporterSamplesTests', () => {
    it('BeetAnGeSample', async () => {
        await MusicXmlImporterTestHelper.testReferenceFile('test-data/musicxml-samples/BeetAnGeSample.xml');
    });

    it('Binchois', async () => {
        await MusicXmlImporterTestHelper.testReferenceFile('test-data/musicxml-samples/Binchois.xml');
    });

    it('BrahWiMeSample', async () => {
        await MusicXmlImporterTestHelper.testReferenceFile('test-data/musicxml-samples/BrahWiMeSample.xml');
    });

    it('BrookeWestSample', async () => {
        await MusicXmlImporterTestHelper.testReferenceFile('test-data/musicxml-samples/BrookeWestSample.xml');
    });

    it('Chant', async () => {
        await MusicXmlImporterTestHelper.testReferenceFile('test-data/musicxml-samples/Chant.xml');
    });

    it('DebuMandSample', async () => {
        await MusicXmlImporterTestHelper.testReferenceFile('test-data/musicxml-samples/DebuMandSample.xml');
    });

    it('Dichterliebe01', async () => {
        await MusicXmlImporterTestHelper.testReferenceFile('test-data/musicxml-samples/Dichterliebe01.xml');
    });

    it('Echigo', async () => {
        await MusicXmlImporterTestHelper.testReferenceFile('test-data/musicxml-samples/Echigo.xml');
    });

    it('FaurReveSample', async () => {
        await MusicXmlImporterTestHelper.testReferenceFile('test-data/musicxml-samples/FaurReveSample.xml');
    });

    it('MahlFaGe4Sample', async () => {
        await MusicXmlImporterTestHelper.testReferenceFile('test-data/musicxml-samples/MahlFaGe4Sample.xml');
    });

    it('MozaChloSample', async () => {
        await MusicXmlImporterTestHelper.testReferenceFile('test-data/musicxml-samples/MozaChloSample.xml');
    });

    it('MozartPianoSonata', async () => {
        await MusicXmlImporterTestHelper.testReferenceFile('test-data/musicxml-samples/MozartPianoSonata.xml');
    });

    it('MozartTrio', async () => {
        await MusicXmlImporterTestHelper.testReferenceFile('test-data/musicxml-samples/MozartTrio.xml');
    });

    it('MozaVeilSample', async () => {
        await MusicXmlImporterTestHelper.testReferenceFile('test-data/musicxml-samples/MozaVeilSample.xml');
    });

    it('Saltarello', async () => {
        await MusicXmlImporterTestHelper.testReferenceFile('test-data/musicxml-samples/Saltarello.xml');
    });

    it('SchbAvMaSample', async () => {
        await MusicXmlImporterTestHelper.testReferenceFile('test-data/musicxml-samples/SchbAvMaSample.xml');
    });

    it('Telemann', async () => {
        await MusicXmlImporterTestHelper.testReferenceFile('test-data/musicxml-samples/Telemann.xml');
    });
});
